src/daml/Daml/Finance/App/Custody/Service.daml

```diff
@@ -76,7 +77,8 @@ template Service
       controller provider
       do
         Custody.CloseAccountRequest{account} <- fetchAndArchive closeAccountRequestCid
-        exercise accountFactoryCid Account.Remove with account
+        (_, ref) <- fetchByKey @Account.R account
```

Comment:
You can also exercise the `Remove` choice on the `Account` interface directly (which archives the ref)